src/android/Dgram.java

```diff
@@ -117,13 +117,14 @@ private void startListening(
 
     private void closeSocket() {
         try {
+            this.closeListener();
```

Comment:
Using this is okay, but given how most java is written in industry using this for the encapsulating class inside a function is actually more confusing.